config.go

```diff
@@ -1077,6 +1089,19 @@ type OCSPConfig struct {
 	ResponderOverrides map[string]string
 }
 
+// CertificateEventData contains contextual information for
+// an obtained, renewed or revoked certificate.
+type CertificateEventData struct {
```

Comment:
```suggestion
type CertificateEvent struct {
```

Reply:
This isn't the actual event, the event is `cert_obtained` etc. This is the data attached to the event.

Reply:
Oh fine. What about `EventData` and then we can add fields that might just be empty depending on the kind of event?

Reply:
True. I wonder if we should change the `emit()` signature to simply take an `Event` struct where the `Name string` is a required field, and the `Data` can be type-asserted... that way the name and data stay together.

Oh and a timestamp might not be a bad idea.

Reply:
Ah right. And it's true, maybe Timestamp isn't super useful. Just seems like a prudent thing to have to document when an event took place in case they want to store or process it or something. Or if event handling is blocked/lagging behind.